src/core/lp_parser.mly

```diff
@@ -0,0 +1,409 @@
+
+// Modifiers of declarations.
+modifier:
+  | CONSTANT { make_pos $loc (P_prop(Terms.Const)) }
+  | INJECTIVE { make_pos $loc (P_prop(Terms.Injec)) }
+  | PROTECTED { make_pos $loc (P_expo(Terms.Protec)) }
+  | PRIVATE { make_pos $loc (P_expo(Terms.Privat)) }
+  | SEQUENTIAL { make_pos $loc (P_mstrat(Terms.Sequen)) }
+  | OPAQUE { make_pos $loc (P_opaq) }
+
+// Converts floats and integers to floats
+float_of_int:
+  | p=FLOAT { p }
+  | n=INT   { float_of_int n }
+
+// Configurations
+config:
+  // Add a builtin: [builtin "0" ≔ zero]
+  | BUILTIN s=STRINGLIT ASSIGN qid=qident { P_config_builtin(s, qid) }
+  // Add a prefix operator: [prefix 1.2 "!" ≔ factorial]
+  | PREFIX p=float_of_int s=STRINGLIT ASSIGN qid=qident
+      { let unop = (s, p, qid) in P_config_unop(unop) }
+  // Add an infix operator: [infix right 6.3 "+" ≔ plus]
+  | INFIX a=ASSOC? p=float_of_int s=STRINGLIT ASSIGN qid=qident
+      {
+        let binop = (s, Option.get Assoc_none a, p, qid) in
+        P_config_binop(binop)
+      }
+  | UNIF_RULE r=unif_rule { P_config_unif_rule(r) }
+
+assert_not:
+  | ASSERT { false }
+  | ASSERT_NOT { true }
+
+query:
+  // Set verbosity level
+  | SET VERBOSE i=INT { make_pos $loc (P_query_verbose(i)) }
+  | SET FLAG s=STRINGLIT b=SWITCH { make_pos $loc (P_query_flag(s,b)) }
+  | SET fl=DEBUG_FLAGS
+      { let (b, s) = fl in make_pos $loc (P_query_debug(b, s)) }
+  | COMPUTE_TYPE t=term
+    { make_pos $loc (P_query_infer(t, {strategy=NONE; steps=None}))}
+  | COMPUTE t=term
+    { make_pos $loc (P_query_normalize(t, {strategy=SNF; steps=None})) }
+  | SET PROVER s=STRINGLIT { make_pos $loc (P_query_prover(s)) }
+  | SET PROVER_TIMEOUT n=INT { make_pos $loc (P_query_prover_timeout(n)) }
+  | k=assert_not ps=arg* TURNSTILE t=term COLON a=term
+    {
+      let t =
+        if ps = [] then t else
+        make_pos ($startpos(ps), $endpos(a)) (P_Abst(ps, t))
+      in
+      let a =
+        if ps = [] then a else
+        make_pos ($startpos(ps), $endpos(a)) (P_Prod(ps, a))
+      in
+      make_pos $loc (P_query_assert(k, P_assert_typing(t, a)))
+    }
+  | k=assert_not ps=arg* TURNSTILE t=term EQUIV a=term
+    {
+      let t =
+        if ps = [] then t else
+        make_pos ($startpos(ps), $endpos(a)) (P_Abst(ps, t))
+      in
+      let a =
+        if ps = [] then a else
+        make_pos ($startpos(ps), $endpos(a)) (P_Abst(ps, a))
+      in
+      make_pos $loc (P_query_assert(k, P_assert_conv(t, a)))
+    }
+  | PRINT qid=qident? { make_pos $loc (P_query_print qid) }
+  | PROOFTERM { make_pos $loc P_query_proofterm }
+
+proof_end:
+  | END { make_pos $loc Syntax.P_proof_end }
+  | ADMIT { make_pos $loc Syntax.P_proof_admit }
+  | ABORT { make_pos $loc Syntax.P_proof_abort }
+
+proof: BEGIN ts=ttactic* pe=proof_end { ts, pe }
+
+inductive:
+  | i=ident COLON t=term ASSIGN VBAR?
+c=separated_list(VBAR, separated_pair(any_ident, COLON, term))
+      { make_pos $loc (fst i, t, c) }
+
+
+// Top level commands
+command:
```

Comment:
Could you please remember me and add in a comment why adding a command terminator is necessary? (You do not require this for tactics.)

Reply:
If we don't do so, Menhir isn't happy. In reality, it's rather rare to not need a terminator, and Earley was more the exception